### PR TITLE
Rt/yj/ruletest 6 1

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
This PR includes 
1. Updated testing JSON file
2. The code updates that reflect the 6-1 RDS

The changes from the 6-2 JSON testing file should be fine because these changes are identical to the ones in the 6-2 PR.

I'd appreciate it if you reviewed this PR. Thank you!

### CODE REVIEW

File: rct229/ruletest_engine/ruletest_jsons/section6/rule_6_1.json [hunk 1]
```diff
@@ -0,0 +1,525 @@
+{
+    "rule-6-1-a": {
+        "Section": 6,
+        "Rule": 1,
+        "Test": "a",
+        "test_description": "Project has one building segment with two spaces. Both lighting power per area and lighting power allowance have been specified. The proposed building lighting power per area equals the maximum value of the proposed building's lighting power allowance and user building's lighting power per area.",
```
Comment (jugonzal07) [inline rct229/ruletest_engine/ruletest_jsons/section6/rule_6_1.json:6]: Hey Yun Joon,

I think the test description is the same for all 6_1 rule jsons. Is it possible to update them to reflect the differences between the tests? That would help me see the intended differences!

Comment (yunjoonjung-PNNL) [reply rct229/ruletest_engine/ruletest_jsons/section6/rule_6_1.json]: Thanks for finding this Juan. I just updated the test description parts.

**Review by jugonzal07 — APPROVED** [review]

**Comment by weilixu** [comment]
The scope of the 6-1 shall be at building segment level. So the calculation should all be moved to building segment as well as the rule assertions.

File: rct229/rule_engine/rule_base.py [hunk 2]
```diff
@@ -605,3 +610,29 @@ def get_fail_msg(self, context, calc_vals=None, data=None):
...
+        context : UserBaselineProposedVals
+            Object containing the contexts for the user, baseline, and proposed RMRs
+        calc_vals : dict or None
+
+        data : An optional data object. It is ignored by this base implementation.
+
+        Returns
+        -------
+        str
+            The message associated with the Pass or Fail case
```
Comment (charliepnnl) [inline rct229/rule_engine/rule_base.py:635]: Shouldn't this be 
```
    The message associated with the Pass case
```
?

File: rct229/rules/section6/section6rule1.py [hunk 3]
```diff
@@ -13,86 +19,164 @@ class Section6Rule1(RuleDefinitionListIndexedBase):
...
-            building_allowable_lighting_power = 0
-            building_design_lighting_power = 0
+        def create_data(self, context, data):
+            building_segment_p = context.proposed
+
+            return {
+                "allowable_LPD_BAM": table_G3_8_lookup(
+                    building_segment_p["lighting_building_area_type"]
+                )["lpd"] if building_segment_p.get("lighting_building_area_type") != None else ZERO.POWER_PER_AREA,
+                "building_area_type_bool": True if building_segment_p.get("lighting_building_area_type") != None else False
```
Comment (charliepnnl) [inline rct229/rules/section6/section6rule1.py:46]: Could shorten line 46 to
```
"building_area_type_bool": building_segment_p.get("lighting_building_area_type") != None
```

**Review by charliepnnl — CHANGES_REQUESTED** [review]
I believe the most nested Rule should be BuildingSegmentRule because the Rule Assertions, Case 1-7 seem to refer to the building segment.

**Comment by yunjoonjung-PNNL** [comment]
I've updated 6-1 code based on the comments and would appreciate it if you could review this PR one more time.

File: rct229/rules/section6/section6rule1.py [hunk 4]
```diff
@@ -14,92 +19,135 @@ class Section6Rule1(RuleDefinitionListIndexedBase):
...
+                else:
+                    return False
+            else:
+                return False
+
+        def get_pass_msg(self, context, calc_vals=None, data=None):
+            allowable_LPD_BAM = calc_vals["allowable_LPD_BAM"]
+            check_BAM_flag = calc_vals["check_BAM_flag"]
+
+            if allowable_LPD_BAM and not check_BAM_flag:
```
Comment (weilixu) [inline rct229/rules/section6/section6rule1.py:135]: Not sure if this is needed - The default message seems to be an empty string

File: rct229/rules/section6/section6rule1.py [hunk 5]
```diff
@@ -14,92 +19,135 @@ class Section6Rule1(RuleDefinitionListIndexedBase):
...
+            elif not allowable_LPD_BAM and not check_BAM_flag:
+                return CASE3_WARNING
+            else:
+                return CASE5_WARNING
+
+        def get_fail_msg(self, context, calc_vals=None, data=None):
+            allowable_LPD_BAM = calc_vals["allowable_LPD_BAM"]
+            check_BAM_flag = calc_vals["check_BAM_flag"]
+
+            if allowable_LPD_BAM and not check_BAM_flag:
```
Comment (weilixu) [inline rct229/rules/section6/section6rule1.py:146]: See comment above.

**Review by weilixu — CHANGES_REQUESTED** [review]

File: rct229/rules/section6/section6rule1.py (hunk 4 above)
Comment (yunjoonjung-PNNL) [reply rct229/rules/section6/section6rule1.py:135]: Fixed.

File: rct229/rules/section6/section6rule1.py (hunk 5 above)
Comment (yunjoonjung-PNNL) [reply rct229/rules/section6/section6rule1.py:146]: Fixed.

**Review by weilixu — APPROVED** [review]
Approved with one small modification

**Pull request merged**